src/lava/proc/conv/models.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.proc.conv.process import Conv
+
+from lava.proc.conv import utils
+
+
+class AbstractPyConvModel(PyLoihiProcessModel):
+    """Abstract template implemetation of PyConvModel."""
+    kernel_size: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+    stride: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+    padding: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+    dilation: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+    groups: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+
+    def run_spk(self):
+        s_in = self.s_in.recv()
```

Comment:
For clarity one should probably define an abstract property for s_in. Since s_in does not exist yet at the object level. This only happens to not throw a linter warning because there exists a class variable with the same name.
But in reality you are only providing an implementation for s_in and a_out in the sub classes.